core/src/test/java/org/apache/iceberg/encryption/DemoKmsClient.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.encryption;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.util.Base64;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+import javax.crypto.SecretKey;
+
+public class DemoKmsClient implements KmsClient {
+
+  public static final String KEY_FILE_PATH_PROP = "demo.kms.client.keyfile.path";
```

Comment:
are these table properties / catalog properties? There are corresponding classes for them at `TableProperties` and `CatalogProperties`.

Reply:
these are "kms specific" parameters, consumed by the custom KmsClient code. This parameter is relatively permanent and is not a secret, so can be set as a table property. But the other one (the password) can't. In this demo, it is passed as a Hadoop parameter, which can a fully in-memory transport; alternatively, it can be also passed via an env variable, etc.